dotCMS/src/main/java/com/dotcms/rest/api/v1/folder/FolderResource.java

```diff
@@ -106,4 +114,94 @@ public final Response loadFolderByURI(@Context final HttpServletRequest httpServ
+        Response response = null;
+        final InitDataObject initData = this.webResource.init(null, httpServletRequest, httpServletResponse, true, null);
+        final User user = initData.getUser();
+        final String uriParam = !uri.startsWith(StringPool.FORWARD_SLASH) ? StringPool.FORWARD_SLASH.concat(uri) : uri;
+        final Host host = APILocator.getHostAPI().findByName(siteName, user, false);
+        //final Folder folder = folderHelper.loadFolderByURI(siteName,user,uriParam);
+        final Folder folder = APILocator.getFolderAPI().findFolderByPath(uriParam, host, user, false);
+        if(host==null || folder==null) {
+            throw new DoesNotExistException("No folder found for "+uri+" on site "+siteName);
+        }
+        CustomFolderView root = getFolderStructure(folder, user);
+        response = Response.ok( new ResponseEntityView(root) ).build();
+        return response;
+    }
+
+    /**
+     * <p>
+     * This method returns a folder structure with their children recursively based on
+     * the folder returned by findFolderByPath
+     * </p>
+     * @param Folder (folder from findFolderByPath) and User (logged in user)
+     * @return CustomFolderView a folder structure with their children recursively
+     */
+    private final CustomFolderView getFolderStructure(Folder folder, User user){
+
+        CustomFolderView customFolder = convertFrom(folder);
+
+        List<CustomFolderView> foldersChildCustoms = new LinkedList<>();
+        List<Folder> children = null;
+        try {
+            children = APILocator.getFolderAPI().findSubFolders(folder, user, false);
+        } catch (Exception e) {
+            Logger.error(this, "Error getting findSubFolders for folder "+folder.getPath(), e);
+        }
+
+        if(children != null && children.size() != 0){
+            for(final Folder child : children){
+                CustomFolderView recursiveFolder = getFolderStructure(child, user);
+                foldersChildCustoms.add(recursiveFolder);
+            }
+        }
+
+        customFolder.setCustomFolders(foldersChildCustoms);
+
+        return customFolder;
+    }
+
+    /**
+     * <p>
+     * This method maps a Folder to a CustomFolderView
+     * </p>
+     * @param Folder
+     * @return CustomFolderView
+     */
+    private CustomFolderView convertFrom(Folder folder){
+        CustomFolderView customFolder = new CustomFolderView(
+                folder.getPath(),
+                folder.getDefaultFileType(),
+                folder.getFilesMasks(),
+                folder.getIDate(),
+                folder.getHostId(),
+                folder.getIdentifier(),
+                folder.getInode(),
+                folder.getModDate(),
+                folder.getName(),
+                folder.isShowOnMenu(),
+                folder.getSortOrder(),
+                folder.getTitle(),
+                folder.getType()
+        );
+        return customFolder;
```

Comment:
Codacy found an issue: [Consider simply returning the value vs storing it in local variable 'customFolder'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5916479)